js/runner.go

```diff
@@ -312,6 +307,77 @@ func (r *Runner) IsExecutable(name string) bool {
 	return exists
 }
 
+// HandleSummary calls the specified summary callback, if supplied.
+func (r *Runner) HandleSummary(ctx context.Context, summary *lib.Summary) (map[string]io.Reader, error) {
+	summaryDataForJS := summarizeMetricsToObject(summary, r.Bundle.Options)
+
+	out := make(chan stats.SampleContainer, 100)
+	defer close(out)
+
+	go func() { // discard all metrics
+		for range out {
+		}
+	}()
+
+	vu, err := r.newVU(0, out)
+	if err != nil {
+		return nil, err
+	}
+
+	handleSummaryFn := goja.Undefined()
+	if exported := vu.Runtime.Get("exports").ToObject(vu.Runtime); exported != nil {
+		fn := exported.Get(consts.HandleSummaryFn)
+		if _, ok := goja.AssertFunction(fn); ok {
+			handleSummaryFn = fn
+		} else if fn != nil && !goja.IsUndefined(fn) && !goja.IsNull(fn) {
```

Comment:
I checked and `Get` returns `nil` if it doesn't exist. Did you intend for `exports.handleSummary = undefined;` (or `null`) to be a valid thing that won't raise an error or were you over zealous in the checks? 

I am not particularly certain it matters either way, to be honest. If someone has defined `handleSummary` it is probably better to use `delete exports.handleSummary`, on the other hand, I don't think we will be hitting it. So the question is mostly questing what was the idea behind it, not changing it one way or another

Reply:
> were you over zealous in the checks?

Likely that, and it makes sense to leave only the `nil` check :+1: